app/src/main/java/ch/epfl/favo/view/tabs/MapsPage.java

```diff
@@ -73,22 +87,88 @@ public View onCreateView(
     setupView();
-    mGpsTracker = new GpsTracker(Objects.requireNonNull(getActivity()).getApplicationContext());
     SupportMapFragment mapFragment =
         (SupportMapFragment) getChildFragmentManager().findFragmentById(R.id.map);
+    mGpsTracker = new GpsTracker(getContext());
     if (mapFragment != null) {
       mapFragment.getMapAsync(this);
     }
   }
 
+
+
+  private void getLocationPermission() {
+    /*
+     * Request location permission, so that we can get the location of the
+     * device. The result of the permission request is handled by a callback,
+     * onRequestPermissionsResult.
+     */
+    if (ContextCompat.checkSelfPermission(Objects.requireNonNull(getActivity()).getApplicationContext(),
+            android.Manifest.permission.ACCESS_FINE_LOCATION)
+            == PackageManager.PERMISSION_GRANTED) {
+      mLocationPermissionGranted = true;
+    } else {
+      ActivityCompat.requestPermissions(Objects.requireNonNull(getActivity()),
+              new String[]{android.Manifest.permission.ACCESS_FINE_LOCATION},
+              PERMISSIONS_REQUEST_ACCESS_FINE_LOCATION);
+    }
+  }
+
+/*
+
+  private void checkPlayServices() {
+    GoogleApiAvailability gApi = GoogleApiAvailability.getInstance();
+    int resultCode = gApi.isGooglePlayServicesAvailable(getActivity());
+    if (resultCode != ConnectionResult.SUCCESS) {
+      gApi.makeGooglePlayServicesAvailable(getActivity());
+    }
+  }
+
+  public void getLocation() throws NoPermissionGrantedException, NoPositionFoundException {
+    getLocationPermission();
+    if (mLocationPermissionGranted) {
+      LocationRequest request = new LocationRequest();
+      request.setPriority(LocationRequest.PRIORITY_BALANCED_POWER_ACCURACY);
+      request.setInterval(15 * 60 * 1000);
+      request.setMaxWaitTime(30 * 60 * 1000);
+      Task<Location> locationResult = mFusedLocationProviderClient.getLastLocation();
+      locationResult.addOnCompleteListener(Objects.requireNonNull(getActivity()), new OnCompleteListener<Location>() {
+        @Override
+        public void onComplete(@NonNull Task<Location> task) {
+          if (task.isSuccessful()) {
+            // Set the map's camera position to the current location of the device.
+            mLocation = task.getResult();
+            GpsTracker.setLastKnownLocation(mLocation);
+          }
+        }
+      });
+    }
+  }
+
+
+  @Override
+  public void onRequestPermissionsResult(int requestCode,
+                                         @NonNull String[] permissions,
+                                         @NonNull int[] grantResults) {
+    mLocationPermissionGranted = false;
+    if (requestCode == PERMISSIONS_REQUEST_ACCESS_FINE_LOCATION) {// If request is cancelled, the result arrays are empty.
+      if (grantResults.length > 0
+              && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
+        mLocationPermissionGranted = true;
+      }
+    }
+  }
+*/
+
```

Comment:
We can remove this whole block of comments if it's not useful anymore

Reply:
actually they are useful, I commented them because I cannot pass the test with them